src/net.cpp

```diff
@@ -1877,8 +1887,10 @@ void CConnman::ThreadMessageHandler()
                 pnode->Release();
         }
 
-        if (fSleep)
-            messageHandlerCondition.timed_wait(lock, boost::posix_time::microsec_clock::universal_time() + boost::posix_time::milliseconds(100));
+        if (fSleep) {
+            std::unique_lock<std::mutex> lock(mutexMsgProc);
+            condMsgProc.wait_until(lock, std::chrono::steady_clock::now() + std::chrono::milliseconds(100));
```

Comment:
I tried to make CThreadInterrupt do this at first (see some of the back+forth with @TheBlueMatt) above. Your suggestion works fine here, I think, but it's not generic due to the lack of accounting for wake conditions. I think it could be done by having wait_until accept an optional predicate, which would be used internally as "while !internal_wake && !pred()".

The compromise was to only use it for plain sleeps. But it's bounced back and forth enough times that I really have no preference. If you'd prefer that, I'm happy to make the change.